eZ/Publish/SPI/FieldType/Exceptions/InvalidIndexDataException.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * File containing the eZ\Publish\SPI\FieldType\Exceptions\InvalidIndexDataException class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\SPI\FieldType\Exceptions;
+
+use Exception;
+
+/**
+ * This Exception is thrown if data given to the indexer are invalid.
+ */
+class InvalidIndexDataException extends Exception
+{
+    /**
+     * @var mixed
+     */
+    protected $fieldId;
+
+    /**
+     * @var int|null
+     */
+    protected $versionNo;
+
+    /**
+     * @var string
+     */
+    protected $languageCode;
+
+    public function __construct($fieldId, $versionNo, $languageCode, Exception $previous = null)
+    {
+        $this->fieldId = $fieldId;
+        $this->versionNo = $versionNo;
+        $this->languageCode = $languageCode;
+
+        $message = sprintf('Field %d in the version %d for %s language can not be indexed', $this->fieldId, $this->versionNo, $this->languageCode);
+        parent::__construct($message, 0, $previous);
```

Comment:
Really small nitpick: we shouldn't use 0 for error code (though I saw we often do). 0 means success (for Mac also, so you can't say I'm biased ;P). Due to lack of better idea I usually use 1. This is of course pure matter of convention.